CMakeLists.txt

```
@@ -22,16 +22,16 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 # add_compile_options(-Wall -Wextra -Wpedantic)
 
-# Enable LTO when possible.
-include(CheckIPOSupported)
-check_ipo_supported(RESULT result OUTPUT output)
-if(result)
-    if(NOT DEFINED CMAKE_INTERPROCEDURAL_OPTIMIZATION)
-        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
-    endif()
-else()
-    message(INFO "IPO is not supported: ${output}")
-endif()
+if (NOT DEFINED CMAKE_INTERPROCEDURAL_OPTIMIZATION)
+    message(CHECK_START "Detecting LTO")
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT CMAKE_INTERPROCEDURAL_OPTIMIZATION OUTPUT LTO_ERROR LANGUAGES C)
```

```suggestion
    check_ipo_supported(RESULT CMAKE_INTERPROCEDURAL_OPTIMIZATION OUTPUT LTO_ERROR)
```
It will be picked from the project language automatically.